fix(yabloc): suppress no viable conversion error

KYabuuchi opened the pull request:
## Description

I suppressed the warnings reported in  https://github.com/autowarefoundation/autoware.universe/issues/7210
Some other warnings still remain. Only the **"no viable conversion error"** has been resolved.

To resolve `no viable conversion` error I made the following change
* Used `static_cast` for explicit type conversion
* Used the direct creation of ColorRGBA with `tier4_autoware_utils::createMarkerColor`

<!-- Write a brief description of this PR. -->

## Tests performed

### Confirm that the error has been resolved by comparing the outputs of clang-tidy. 

Below, I have provided links to the previous clang-tidy's outputs and those generated by this PR.

* before this PR: https://github.com/autowarefoundation/autoware.universe/actions/runs/9266220660/artifacts/1543408022
(produced by [here]( https://github.com/autowarefoundation/autoware.universe/actions/runs/9266220660/job/25489950665))
* after this PR: https://github.com/autowarefoundation/autoware.universe/actions/runs/9395421937/artifacts/1573979064
(produced by [here]( https://github.com/autowarefoundation/autoware.universe/actions/runs/9395421937/job/25874812761?pr=7299))


### Confirmed that yabloc works as conventional in logging_simulator

```
ros2 launch autoware_launch logging_simulator.launch.xml \
    map_path:=$HOME/Maps/nishishinjuku \
    vehicle_model:=sample_vehicle \
    sensor_model:=awsim_sensor_kit \
    pose_source:=yabloc
```

* [conventional rosbag](https://drive.google.com/file/d/1zKTGRH4lD-wptpOdNCgpiPfGRDP0XrUm/view?usp=drive_link)
   * [new rosbag (TIER IV INTERNAL USER ONLY)](https://drive.google.com/file/d/1iEostJWDNQhz_uMDckN1qjDEf81f8Xw2/view?usp=drive_link) which has autoware_msg conveted from autoware_auto_msg  
* [map](https://github.com/tier4/AWSIM/releases/download/v1.1.0/nishishinjuku_autoware_map.zip) : This link was quoted from [AWSIM tutorial](https://tier4.github.io/AWSIM/GettingStarted/QuickStartDemo/#start-the-demo).


## Effects on system behavior

<!-- Describe how this PR affects the system behavior. -->

This PR does not change the system behavior.

## Interface changes

Interface does not change.
<!-- Describe any changed interfaces, such as topics, services, or parameters, including debugging interfaces -->

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

SakodaShintaro commented:
I don't really understand what the change means, but I don't think it's a problem.

Small comments

(1) The CI check "semantic-pull-request" is failing for some reason (is it because there are spaces on both sides of the ":"?)

(2) There is a "()" in the "Tests performed" section in PR description.

![image](https://github.com/autowarefoundation/autoware.universe/assets/28504069/5bdf35d3-b92f-45da-a10b-f142de2ec477)

KYabuuchi commented:
@SakodaShintaro Thank you for the review.  :bow:
Here are the details of the changes I made (I also added this to the PR description):

* Used `static_cast` for explicit type conversion
* Used the direct creation of `ColorRGBA` with `tier4_autoware_utils::createMarkerColor`

Also, I fixed the issue of semantic-pull-request, and the typo "()" in the PR description.

SakodaShintaro reviewed (approved):
Thank you for the fix!
Looks Good To Me
